src/python/geoclaw/util.py

```diff
@@ -141,3 +150,121 @@ def inv_haversine(d,x1,y1,y2,Rsphere=Rearth,units='degrees'):
     if units=='degrees':
         dx = dx * RAD2DEG
     return dx
+
+
+def fetch_noaa_tide_data(station, begin_date, end_date, time_zone='GMT',
+                         datum='STND', units='metric', cache_dir='_cache'):
+    r"""Fetch water levels and tide predictions at given NOAA tide station.
+
+    The data is returned in 6 minute intervals between the specified begin and
+    end dates/times.  A complete specification of the NOAA CO-OPS API for Data
+    Retrieval used to fetch the data can be found at:
+
+        https://tidesandcurrents.noaa.gov/api/
+
+    By default, retrieved data is cached in the subdirectory '_cache' of the
+    current working directory.
+
+    :Required Arguments:
+      - station (string): 7 character station ID
+      - begin_date (datetime): start of date/time range of retrieval
+      - end_date (datetime): end of date/time range of retrieval
+
+    :Optional Arguments:
+      - time_zone (string): see NOAA API documentation for possible values
+      - datum (string): see NOAA API documentation for possible values
+      - units (string): see NOAA API documentation for possible values
+      - cache_dir (string): directory to use for caching retrieved data
+
+    :Returns:
+      - date_time (numpy.ndarray): times corresponding to retrieved data
+      - water_level (numpy.ndarray): preliminary or verified water levels
+      - prediction (numpy.ndarray): tide predictions
+    """
+    def fetch(product, expected_header, col_idx, col_types):
+        noaa_params = get_noaa_params(product)
+        cache_path = get_cache_path(product)
+
+        # use cached data if available
+        if os.path.exists(cache_path):
+            print('Using cached {} data for station {}'.format(
+                product, station))
+            return parse(cache_path, col_idx, col_types, header=True)
+
+        # otherwise, retrieve data from NOAA and cache it
+        print('Fetching {} data from NOAA for station {}'.format(
```

Comment:
Might suggest allowing this to be suppressed by adding a `verbose` keyword argument that defaults to `True`.

Reply:
Sure, fixed.